packages/typescript-estree/src/node-utils.ts

```diff
@@ -335,13 +335,20 @@ export function isJSXToken(node: ts.Node): boolean {
  */
 export function getDeclarationKind(
   node: ts.VariableDeclarationList,
-): 'const' | 'let' | 'var' {
+): 'const' | 'let' | 'var' | 'using' | 'await using' {
   if (node.flags & ts.NodeFlags.Let) {
     return 'let';
   }
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-enum-comparison
+  if ((node.flags & ts.NodeFlags.AwaitUsing) === ts.NodeFlags.AwaitUsing) {
```

Comment:
[Aside] Amusing: https://github.com/typescript-eslint/typescript-eslint/issues/6909. (not requesting changes, just amused)